CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+# Contributing an SCM provider
+An SCM provider support has to be provided by adding new maven modules to the wsmaster directory.
+
+## Implementing che-core-api-oauth-<SCM provider name> module
+This module is responsible for Oauth requests to the SCM provider and contans next implementations:
+1. `<SCM provider name>OAuthAuthenticator` contains specific implementation of Oauth token requestand authentication endpoint.
+2. `<SCM provider name>OAuthAuthenticatorProvider` a provider of the `OAuthAuthenticator`.
+
+## Implementing coresponding che-core-api-factory-<SCM provider name> module
+This module is responsible for API operations of the specific SCM provider.
+1. `<SCM provider name>ApiClient` contains all necessary HTTP API calls like `getUser()`.
+2. `<SCM provider name>AuthorisingFileContentProvider` overrides the common `AuthorisingFileContentProvider` to define the specific `isPublicRepository()` function.
+3. `<SCM provider name>FactoryParameterResolver` validates SCM URL if it corresponds to the SCM provider. Also Provides specific Factory Parameters resolver for the SCM repository.
+4. `<SCM provider name>PersonalAccessTokenFetcher` fetches Personal Access Token from the SCM provider by reqesting the OAuth API of the provider. Validates the token by comparing the token access scopes with the predefined scope list.
+5. `<SCM provider name>FileResolver` Implementation of a resolver that can return particular file content from specified SCM repository.
+6. `<SCM provider name>URLParser` Parses the string representation of the URL to an SCM specific object 
+7. `<SCM provider name>UserDataFetcher` Implementation of a resolver that can return particular file content from specified SCM repository.
+
+# Build requirements
+- Apache Maven 3.9 or later
+- JDK 11
+- Podman or Docker (required for running integration tests)
+A Che workspace environment allows to build the image internaly, using the workspace terminal.
+
+# Sources build
+Run `mvn clean install` to build. Activate a faster profile build by adding `-Pfast`.
+
+# Image build and push
+1. Go to the `dockerfiles` directory.
+2. Run `./build.sh`.
+3. Tag the **che-server** image with your account: `docker tag quay.io/eclipse/che-server:next <docker registry>/<your account>/che-server:next`.
+4. Push the **che-server** image to your account: `docker push <docker registry>/<your account>/che-server:next`.
+
+# Start and debug
```

Comment:
Could we also add a `Developing within Eclipse Che` section ? similar to https://github.com/eclipse-che/che-dashboard/blob/main/CONTRIBUTING.md#developing-within-eclipse-che

Reply:
done